### PR TITLE
Multiple improvements to docs.

### PR DESCRIPTION
**Pull request by YashJipkate**

### CODE REVIEW

File: cpp/src/arrow/adapters/arrow-rados-cls/README.md [hunk 1]
```diff
@@ -76,3 +76,7 @@ docker-compose run --service-ports ubuntu-cls-demo
 # Development
 
 Check out these [instructions](docs/contributing.md) for setting up a local development environment.
+
+# Publication
+
+[Towards an Arrow-native Storage System](https://arxiv.org/pdf/2105.09894.pdf)
```
Comment (JayjeetAtGithub) [inline cpp/src/arrow/adapters/arrow-rados-cls/README.md:82]: Can we put it in a numbered list ?

**Review by JayjeetAtGithub — CHANGES_REQUESTED** [review]
Thanks, @YashJipkate . Looks Good. Can you link this PR to the issues that it is fixing?

**Comment by YashJipkate** [comment]
I have referenced the issues in the commit messages which should close the respective issues when merged.

File: cpp/src/arrow/adapters/arrow-rados-cls/README.md (hunk 1 above)
Comment (YashJipkate) [reply cpp/src/arrow/adapters/arrow-rados-cls/README.md:82]: But there is just one, so a list would be unnecessary in my opinion. When we add more publications, maybe then a list would be more appropriate, don't you think?

Comment (JayjeetAtGithub) [reply cpp/src/arrow/adapters/arrow-rados-cls/README.md:82]: We are about to add more actually, like from our collaborators from Leiden univ.

Comment (JayjeetAtGithub) [reply cpp/src/arrow/adapters/arrow-rados-cls/README.md:82]: Also rename to `Publications` (prural)

Comment (YashJipkate) [reply cpp/src/arrow/adapters/arrow-rados-cls/README.md]: Done.

**Pull request merged**